libcontainer/rootfs_linux.go

```diff
@@ -400,7 +400,29 @@ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
 			return err
 		}
 		// Selinux kernels do not support labeling of /proc or /sys
-		return mountPropagate(m, rootfs, "", nil)
+		err = mountPropagate(m, rootfs, "", nil)
+		if err != nil && m.Device == "sysfs" {
+			/* If we are running in a user namespace, just bind mount /sys if creating
+			   sysfs failed.  */
+			if !userns.RunningInUserNS() {
+				return err
+			}
+			m.Source, m.Device = "/sys", "bind"
+			m.Flags = unix.MS_BIND | unix.MS_REC | unix.MS_NODEV | unix.MS_NOSUID | unix.MS_NOEXEC
+			if err = utils.WithProcfd(rootfs, m.Destination, func(procfd string) error {
+				return mount(m.Source, m.Destination, procfd, m.Device, uintptr(m.Flags), label.FormatMountLabel(m.Data, mountLabel))
+			}); err != nil {
+				return err
+			}
+			m.RecAttr = &unix.MountAttr{
+				Attr_set: unix.MOUNT_ATTR_RDONLY,
```

Comment:
I think it should be set to RO only if the original flags had RDONLY flag set.

Reply:
If `MOUNT_ATTR_RDONLY` is set only when `RDONLY` is originally set, there will be security issues when `/sys/fs/cgroup` is not set in config.json.

Maybe it can be like the modification of crun, only when `RDONLY` is originally set, then `sysfs` => `/sys`